### ISSUE TITLE
The difference between T2T and opennmt-tf

### ISSUE THREAD

**Issue by BigFishMaster** [OpenNMT/OpenNMT-tf#394]
With the same training data, T2T-based Transformer can reach 3.3 softmax loss, while opennmt-tf-based Transformer has bigger loss( > 4.0). I compare the hyper-parameters to keep the initialization same. Cloud you tell me which problem may I meet? The following is my parameters.
```

params:
    average_loss_in_time: true
    label_smoothing: 0.1
    optimizer: AdamOptimizer
    optimizer_params:
        beta1: 0.9
        beta2: 0.998
    learning_rate: 2.0
    decay_type: noam_decay_v2
    decay_params:
        model_dim: 512
        warmup_steps: 8000
    gradients_accum: 1
    beam_width: 4
    length_penalty: 1
    minimum_decoding_length: 0
    maximum_iterations: 50


train:
    effective_batch_size: null
    batch_size: 4096
    batch_type: tokens
    maximum_features_length: 50
    maximum_labels_length: 50
    sample_buffer_size: 100000
    save_summary_steps: 100
    train_steps: 1000000
    keep_checkpoint_max: 8
    average_last_checkpoints: 8
    bucket_width: 8
    num_threads: 8
    prefetch_buffer_size: 4
    save_checkpoints_steps: 5000
    save_checkpoints_secs: null


eval:
    batch_size: 32
    num_threads: 1
    prefetch_buffer_size: 1
    external_evaluators: [BLEU, ROUGE]
    save_eval_predictions: true
    eval_delay: 1200
    exporter: last


infer:
    batch_size: 32
    bucket_width: 5
    num_threads: 1
    prefetch_buffer_size: 1
    n_best: 1
    with_scores: false


score:
    batch_size: 64
```

**Comment by guillaumekln** [comment]
Are you comparing the training loss? T2T applies an additional normalization constant:

https://github.com/tensorflow/tensor2tensor/blob/a1a6404c829f31d46a0402772d9e2ee4b125d430/tensor2tensor/layers/common_layers.py#L2020

You should compare the evaluation loss or the metric of your task (e.g. BLEU).

**Comment by BigFishMaster** [comment]
@guillaumekln Thx. The 'normalizing' may be the keypoint.  In my task, the vocab size is between [30k, 40k], that value is about 1.35. I try this normalization latter.

**Comment by valentinmace** [comment]
@BigFishMaster Could you tell us (when your training is over), if there's still a difference between T2T and opennmt-tf transformer (in BLEU score for example) ?

**Comment by guillaumekln** [comment]
It should be on par with Tensor2Tensor in terms of BLEU score.
